app/helpers/route_overrides_helper.rb

```diff
@@ -0,0 +1,60 @@
+module RouteOverridesHelper
+  class << self
+    private
+
+    # @param format [Symbol] Either :singular or :plural
+    def mapping(format, options)
+      if format == :singular
+        from = options[:from].to_s.singularize
+        to = options[:to].to_s.singularize
+      else
+        from = options[:from].to_s.pluralize
+        to = options[:to].to_s.pluralize
+      end
+      mapping = {}
+      mapping[from] = to
+      mapping
+    end
+
+    def mapping_for(options)
+      should_map_singular = options[:singular].nil? || options[:singular]
+      should_map_plural = options[:plural].nil? || options[:plural]
+
+      singular_mapping = should_map_singular ? mapping(:singular, options) : {}
+      plural_mapping = should_map_plural ? mapping(:plural, options) : {}
+      singular_mapping.merge(plural_mapping)
+    end
+
+    def override_route_helpers_with_mapping(mapping)
+      ['_path', '_url'].each do |suffix|
+        ['', 'new_', 'edit_'].each do |prefix|
+          mapping.each do |from, to|
+            define_method(prefix + from + suffix) do |*forwarded_args|
+              send(prefix + to + suffix, *forwarded_args)
+            end
+          end
+        end
+      end
+    end
+
+    # Override route helper methods e.g. to remove the namespacing in
+    # the model class.
+    #
+    # @param from [Symbol, String] The route helper to be overridden. This helper could be generated
+    #        by a form helper link_to but is not actually created from the route setup.
+    # @param to [Symbol, String] The correct route to be used which is created by the route setup.
+    # @param singular [Boolean] Set to +false+ to ignore the helper methods in singular form
+    #        e.g. edit_course_achievement_path
+    # @param plural [Boolean] - Set to +false+ to ignore the helper methods in plural form
+    #        e.g. course_achievements
+    def map_route(from:, to:, singular: true, plural: true)
```

Comment:
Actually I think you can refactor this method to make more sense, there's no need to be a direct break of my long function.

Sorry, I think I've learnt a bit more since I wrote that function...